builtin/settingtypes.txt

```diff
@@ -710,8 +710,9 @@ fall_bobbing_amount (Fall bobbing factor) float 0.03
 #    -    sidebyside: split screen side by side.
 #    -    crossview: Cross-eyed 3d
 #    -    pageflip: quadbuffer based 3d.
+#    -    secondstage: second rendering stage for improved visual effects
 #    Note that the interlaced mode requires shaders to be enabled.
-3d_mode (3D mode) enum none none,anaglyph,interlaced,topbottom,sidebyside,crossview,pageflip
+3d_mode (3D mode) enum none none,anaglyph,interlaced,topbottom,sidebyside,crossview,pageflip,secondstage
```

Comment:
I hope this is temporary - post-processing doesn't really fit into 3D mode as a separate mode. It should rather be applied to all 3D modes, ideally.

Reply:
I see your point, but that would be relatively hard to implement...

Reply:
I'm just suggesting the architectural implications of this approach - if post-processing is specified as a 3D mode, it won't be possible to post-process other 3D modes. :)

Maybe just keep this as a separate setting or a shader option? Sort of like "Tone mapping", "Bump mapping", etc.

Reply:
It is definitely possible applying the same technique *before* merging stereo images. For many modes, it can also be applied *instead* of plain merging, but that needs more care. Not sure how to fit that into the current code, though.

Reply:
If I understand the code correctly, with stereo 3D two images are rendered in [RenderingCoreStereo::renderBothImages](https://github.com/minetest/minetest/blob/7be082f9a8bb6bd46c226d7ef4c42f0fd9fe7314/src/client/render/stereo.cpp#L52), whereas without stereo 3D, a single image is rendered in [RenderingCorePlain::drawAll](https://github.com/minetest/minetest/blob/7be082f9a8bb6bd46c226d7ef4c42f0fd9fe7314/src/client/render/plain.cpp#L70).
I think those two methods are the right places for the postprocessing drawing. In the stereo case, postprocessing needs to be performed two times, but the same buffers can be used, so a bit duplicate code may be tolerated here. The buffer initializations can probably be added to the plain and stereo initialization with a test if a postprocessing setting is enabled.